pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -274,13 +274,20 @@ allroutes:
 			}
 			break allroutes // we have a rule with catch all match prefix: /. Other rules are of no use
 		} else {
-			for _, match := range http.Match {
+			catchall := false
+			if match := catchAllMatch(http); match != nil {
+				// We have a catch all route - see if it is valid and if yes, do not build other routes.
```

Comment:
nit: we have a catch all match block in a route. If it is valid, skip building routes for other matches in this rule element

Reply:
fixed